### PR TITLE
feat(apps): Add description & permissions to db

### PR DESCRIPTION
**Pull request by deshmukhmayur**
# Resolves 1479

# Explain the feature/fix

Added additional fields to the App Database schema for description and permissions.

## Does this PR introduce a breaking change

Yes.

There is a schema change for the databases field, but it shouldn't affect anything since it was not being used yet.

## Screenshots

N/A

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/apps-service/src/modules/apps/resolver.ts [hunk 1]
```diff
@@ -101,13 +102,22 @@ export default <IResolvers<App, IAppsContext>>{
...
+
+      if ( isEmpty( database.permissions ) ) {
+        /* TODO: Add default users from the app permission model */
+        database.permissions = { admins: [ 'user:' + app.ownerId ], users: [ 'user:' + app.ownerId, 'op-users' ] }
+      }
 
       try {
         /* Create the database on couchdb */
-        await createDatabase( databaseName );
+        await createDatabase( database.name );
```
Comment (ghanlohar) [inline packages/apps-service/src/modules/apps/resolver.ts:118]: @deshmukhmayur How we are setting the description?

File: packages/apps-service/src/modules/apps/schema.gql.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ type Mutation {
 
   createAppDatabase(id: ID!, databaseName: String!): App
```
Comment (ghanlohar) [inline packages/apps-service/src/modules/apps/schema.gql.ts:15]: We need to accept the `description` and `permissions` as the input in the mutation for `createAppDatabase`.

**Review by ghanlohar — CHANGES_REQUESTED** [review]

File: packages/apps-service/src/modules/apps/resolver.ts (hunk 1 above)
Comment (deshmukhmayur) [reply packages/apps-service/src/modules/apps/resolver.ts:118]: This is the service call to couchdb. The description is not needed for this.

File: packages/apps-service/src/modules/apps/schema.gql.ts (hunk 2 above)
Comment (deshmukhmayur) [reply packages/apps-service/src/modules/apps/schema.gql.ts]: Updated.

**Review by ghanlohar — APPROVED** [review]

**Review by riginoommen — APPROVED** [review]

**Pull request merged**